dataclients/kubernetes/hosts.go

```diff
@@ -13,14 +13,18 @@ func rxDots(h string) string {
 	return strings.Replace(h, ".", "[.]", -1)
 }
 
+func appendOptionalPortRegex(host string) string {
+	return fmt.Sprintf("%s(:[0-9]+)?", host)
+}
+
 func createHostRx(h ...string) string {
 	if len(h) == 0 {
 		return ""
 	}
 
 	hrx := make([]string, len(h))
 	for i := range h {
-		hrx[i] = rxDots(h[i])
```

Comment:
rxDots() is not used anymore, please also cleanup this function definition in the PR, thanks

Reply:
The function is still used but it's wrapped as the argument for the new function.

Reply:
Oh my eyes :D

Reply:
We could move `generateHostRegexp` here from ingress.go and reuse

Reply:
@girishc13 Lets reuse `generateHostRegexp` here so that we use a single way to create hostname regexp

Reply:
@girishc13 this seems to last and missing piece to do before we can merge.

Reply:
I took a quick look , there is one issue that's preventing me from directly using the `generateHostRegexp` method.
This method is using this statement for format the hosts:
```
return fmt.Sprintf("^(%s)$", strings.Join(hrx, "|"))
```
The `generateHostsRegexp` is formatting the host as
```
return "^" + strings.Replace(host, ".", "[.]", -1) + "(:[0-9]+)?$"
```

There is a mismatch in the capture group. I will take a deeper look but please feel free to provide suggestions.

Reply:
One idea is to extend the `generateHostRegexp` to accept two more arguments for begin expression and end expression like
```
generateHostRegexp(beginExp, host, endExp) {
   return beginExp + strings.Replace(host, ".", "[.]", -1) + "(:[0-9]+)?" + endExp
}
```

Reply:
There is already `createHostRx` that should do what we need, something like this maybe:
```diff
diff --git a/dataclients/kubernetes/eastwest.go b/dataclients/kubernetes/eastwest.go
index d20f814d..cecb0a37 100644
--- a/dataclients/kubernetes/eastwest.go
+++ b/dataclients/kubernetes/eastwest.go
@@ -16,7 +16,7 @@ func createEastWestRouteIng(eastWestDomain, name, ns string, r *eskip.Route) *es
 		return nil
 	}
 	ewR := *r
-	ewR.HostRegexps = []string{generateHostRegexp(name + "." + ns + "." + eastWestDomain)}
+	ewR.HostRegexps = []string{createHostRx(name + "." + ns + "." + eastWestDomain)}
 	ewR.Id = eastWestRouteID(r.Id)
 	return &ewR
 }
diff --git a/dataclients/kubernetes/hosts.go b/dataclients/kubernetes/hosts.go
index 147aa8c2..9a24e884 100644
--- a/dataclients/kubernetes/hosts.go
+++ b/dataclients/kubernetes/hosts.go
@@ -1,7 +1,6 @@
 package kubernetes
 
 import (
-	"fmt"
 	"net/url"
 	"regexp"
 	"strings"
@@ -9,25 +8,19 @@ import (
 	"github.com/zalando/skipper/eskip"
 )
 
-func rxDots(h string) string {
-	return strings.Replace(h, ".", "[.]", -1)
-}
-
-func appendOptionalPortRegex(host string) string {
-	return fmt.Sprintf("%s(:[0-9]+)?", host)
-}
-
-func createHostRx(h ...string) string {
-	if len(h) == 0 {
+func createHostRx(hosts ...string) string {
+	if len(hosts) == 0 {
 		return ""
 	}
-
-	hrx := make([]string, len(h))
-	for i := range h {
-		hrx[i] = appendOptionalPortRegex(rxDots(h[i]))
+	hrx := make([]string, len(hosts))
+	for i, host := range hosts {
+		hrx[i] = strings.Replace(host, ".", "[.]", -1) + "(:[0-9]+)?"
 	}
-
-	return fmt.Sprintf("^(%s)$", strings.Join(hrx, "|"))
+	// single host regexp could be made a bit cleaner
+	// 	if len(hrx) == 1 {
+	// 		return "^" + hrx[0] + "$"
+	// 	}
+	return "^(" + strings.Join(hrx, "|") + ")$"
 }
 
 // hostCatchAllRoutes creates catch-all routes for those hosts that only have routes with
diff --git a/dataclients/kubernetes/ingress.go b/dataclients/kubernetes/ingress.go
index ce8af574..7953671a 100644
--- a/dataclients/kubernetes/ingress.go
+++ b/dataclients/kubernetes/ingress.go
@@ -177,7 +177,7 @@ func convertPathRule(
 
 	var hostRegexp []string
 	if host != "" {
-		hostRegexp = []string{generateHostRegexp(host)}
+		hostRegexp = []string{createHostRx(host)}
 	}
 	svcPort := prule.Backend.ServicePort
 	svcName := prule.Backend.ServiceName
@@ -245,10 +245,6 @@ func convertPathRule(
 	return r, nil
 }
 
-func generateHostRegexp(host string) string {
-	return "^" + strings.Replace(host, ".", "[.]", -1) + "(:[0-9]+)?$"
-}
-
 func setTraffic(r *eskip.Route, svcName string, weight float64, noopCount int) {
 	// add traffic predicate if traffic weight is between 0.0 and 1.0
 	if 0.0 < weight && weight < 1.0 {
@@ -489,7 +485,7 @@ func (ing *ingress) addSpecRule(ic ingressContext, ru *definitions.Rule) error {
 	// it is a regexp, would be better to have exact host, needs to be added in skipper
 	// this wrapping is temporary and escaping is not the right thing to do
 	// currently handled as mandatory
-	host := []string{generateHostRegexp(ru.Host)}
+	host := []string{createHostRx(ru.Host)}
 	// update Traffic field for each backend
 	computeBackendWeights(ic.backendWeights, ru)
 	for _, prule := range ru.Http.Paths {
```